compiler/src/dotty/tools/dotc/transform/PruneErasedDefs.scala

```diff
@@ -34,19 +35,26 @@ class PruneErasedDefs extends MiniPhase with SymTransformer { thisTransform =>
 
   override def transformApply(tree: Apply)(implicit ctx: Context): Tree =
     if (tree.fun.tpe.widen.isErasedMethod)
-      cpy.Apply(tree)(tree.fun, tree.args.map(arg => ref(defn.Predef_undefined)))
+      cpy.Apply(tree)(tree.fun, tree.args.map(trivialErasedTree))
     else tree
```

Comment:
In PostTyper, we also have logic to erase arguments. Is there a duplication of code?

Reply:
We keep in TASTy the contents of the erased trees. These are erased here. In PostTyper we only remove contents of erased definitions that were generated by inlining code (i.e. needed for typing). I will recheck what happens if we don't remove them in PostTyper, but fur sure the TASTy will increase.